src/ristretto/pedersen/extended_commitment_factory.rs

```diff
@@ -0,0 +1,415 @@
+                let k_i = vec![RistrettoSecretKey::random(&mut rng); extension_degree as usize];
+                for i in 0..extension_degree as usize {
+                    k_sum[i] = &k_sum[i] + &k_i[i];
+                }
+                let c = factory.commit(&k_i, &v).unwrap();
+                c_sum = &c_sum + &c;
+                commitments.push(c);
+            }
+            assert!(factory.open(&k_sum, &v_sum, &c_sum).unwrap());
+            assert_eq!(c_sum, commitments.iter().sum());
+        }
+    }
+
+    #[test]
+    fn serialize_deserialize() {
+        let mut rng = rand::thread_rng();
+        for extension_degree in EXTENSION_DEGREE {
+            let factory = ExtendedPedersenCommitmentFactory::new_with_extension_degree(extension_degree).unwrap();
+            let k = vec![RistrettoSecretKey::random(&mut rng); extension_degree as usize];
+            let c = factory.commit_value(&k, 420).unwrap();
+            // Base64
+            let ser_c = c.to_base64().unwrap();
+            let c2 = PedersenCommitment::from_base64(&ser_c).unwrap();
+            assert!(factory.open_value(&k, 420, &c2).unwrap());
+            // MessagePack
+            let ser_c = c.to_binary().unwrap();
+            let c2 = PedersenCommitment::from_binary(&ser_c).unwrap();
+            assert!(factory.open_value(&k, 420, &c2).unwrap());
+            // Invalid Base64
+            assert!(PedersenCommitment::from_base64("bad@ser$").is_err());
+        }
+    }
+
+    #[test]
+    fn derived_methods() {
+        for extension_degree in EXTENSION_DEGREE {
+            let factory = ExtendedPedersenCommitmentFactory::new_with_extension_degree(extension_degree).unwrap();
+            let k = vec![RistrettoSecretKey::from(1024); extension_degree as usize];
+            let c1 = factory.commit_value(&k, 2048).unwrap();
+            let mut hasher = DefaultHasher::new();
+            c1.hash(&mut hasher);
+            let c2 = c1.clone();
+            let c3 = factory.commit_value(&k, 2049).unwrap();
+
+            // Test 'clone`
+            assert_eq!(c1, c2);
+            assert_ne!(c2, c3);
+
+            // Note! 'Ord' and 'PartialOrd' are undefined for extended commitments with extension degree other than Zero
+            match extension_degree {
+                ExtensionDegree::Zero => {
+                    // Test 'Debug'
+                    assert_eq!(
+                        format!("{:?}", c1),
+                        "HomomorphicCommitment(9801c7785217e0c973e9b85508c6eebcb74b257a6f825630e17282b81b7fcd78)"
+                    );
+                    // Test hashing
+                    let result = format!("{:x}", hasher.finish());
+                    assert_eq!(&result, "1ef11e8d243c886e");
+                    // test 'Ord' and 'PartialOrd'
+                    assert!(c2 > c3);
+                    assert!(c3 < c2);
+                    assert!(matches!(c2.cmp(&c3), std::cmp::Ordering::Greater));
```

Comment:
In case there's any misunderstanding, point ordering is lexicographic on canonical byte representations, and has no algebraic significance of any kind (in the group structure). Performing algebraic operations on points (in this example, committing to a larger or smaller value with the same mask) is not guaranteed or expected to lead to any specific ordering of them. Ordering is useful for operations like sorting for hashing purposes, where a given set of points needs a canonical ordering.